mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -596,7 +596,7 @@ def ask_selection(self, options, dialog='',
                 if self.voc_match(resp, 'last'):
                     resp = options[-1]
                 else:
-                    num = extract_number(resp, self.lang, ordinals=True)
+                    num = extract_number(resp, ordinals=True, lang=self.lang)
```

Comment:
are there circumstances under which `self.lang != lingua_franca.get_default_lang()`?

Reply:
Yeah I thought about this briefly when changing it, there's at least one more instance in that file to `join_list()`

My thinking was that the `ask_selection()` method has no `lang` parameter and will therefore always use `self.lang` to select it's dialog. So we should definitely have `join_list(options, "or", lang=self.lang)` as it will be part of the dialog. `if self.lang != default_lingua_franca_lang` then any dialog that requires some `lf.format` is going to be multilingual and likely not what was intended.

For parsing it makes more sense as it will return a computer readable object but I still think it would be better to use the `lang` argument on a method rather than setting the default language globally.

In short - I don't think they should be different, but there's nothing programmatically stopping that from happening. So my gut is saying it's safer to be explicit here.

Reply:
Good call, and good catch!